register-azure-devops-npm-auth.ps1

```diff
@@ -0,0 +1,138 @@
+<#
+  .SYNOPSIS
+  Register azure-devops-npm-auth tool as an app in Azure AD and grants it admin consent
+
+  .DESCRIPTION
+  In practice, in order to use azure-devops-npm-auth tool to authenticate to an Azure artifacts private npm feed
+  you likely need admin consent granted to it so that it has the permissions to work on behalf of you the developer.
+  For that to happen (in a reliable way), the tool should be registered as an app in your Azure AD tenant that
+  is connected to the Azure devops organisation/project that hosts Azure Artifacts. This script does that.
+
+  Specifically the script:
+  * creates an Azure AD App registration and associated service principal
+  * grants admin consent to the azure-devops-npm-auth tool the following permissions:
+    - Microsoft Graph > User.Read
+    - Azure Devops > user_impersonation
+
+  Permissions to run this script: Azure AD 'Global administrator' RBAC role
+  Tools required to run this script: azure-cli (see https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
+
+  For more information see https://www.npmjs.com/package/azure-devops-npm-auth
+
+  .EXAMPLE
+  ./register-azure-devops-npm-auth.ps1 -Login -InfA Continue
+
+  Description
+  -----------
+  Trigger an interactive login, printing the INFO level log messages to the console
+
+#>
+
+
+[CmdletBinding()]
+param(
+    [switch] $Login
+)
+begin {
+    Set-StrictMode -Version 'Latest'
+    $callerEA = $ErrorActionPreference
+    $ErrorActionPreference = 'Stop'
+
+    function Invoke-Exe {
+        param(
+            [Parameter(Mandatory)]
+            [ScriptBlock] $ScriptBlock
+        )
+        Write-Verbose "Executing: $ScriptBlock"
+        Invoke-Command $ScriptBlock
+        if ($LASTEXITCODE -ne 0) {
+            throw "Command failed with exit code $LASTEXITCODE; Cmd: $ScriptBlock"
+        }
+    }
+
+}
+process {
+    try {
+
+        if ($Login) {
+            Write-Information 'Connecting to Azure AD Account...'
+            Invoke-Exe { az login } | Out-Null
```

Comment:
isn't it better to use the PowerShell Az-Module instead?

Reply:
The Az modules are REALLY not complete yet. I found that out the hard way trying to automate app role assignment. So once bitten twice shy!